storage/pebble/register.go

```diff
@@ -0,0 +1,181 @@
+package pebble
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/pebble/registers"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/onflow/atree"
+
+	"github.com/onflow/flow-go/fvm/evm/types"
+)
+
+var (
+	// MinLookupKeyLen defines the minimum length for a valid lookup key
+	//
+	// Lookup keys use the following format:
+	//     [key] / [height]
```

Comment:
Yes, the owner can technically omitted, since all registers are stored under the same owner (EVM root address). 

I wonder how much do we save, because I think pebble has compression on keys, which means the common prefix on keys will all be compressed, and the difference would be very minimal IMO.

However, we would lose some benefit if we store it differently from execution node:

1. we could run/debug gateway code with EN's register index data, because they are storing the same registers in the same format. 
2. register related tools made for EN in the future can't  be directly used by the gateway 
3. mainnet database/snapshot can't be used by testnet by mistake, because the root owner is different.
4. we might want to compare the full set of registers with execution node to see if they are consistent. If the schema is the same, we could simply dump all registers to a file using the same tool, and compare them. 
5. we could reuse the pruning module from flow-go without changing anything if we need one day. 

If the saving isn't significant, then I would suggest to keep the owner. 

Maybe you can try with a benchmark, that saves tons of different registers and compare the disk size.

Reply:
> 1. we could run/debug gateway code with EN's register index data, because they are storing the same registers in the same format.

In theory yes, but the data will not be 100% the same. The reason for this is that on the EN the EVM storage account also contains some extra information not just EVM storage. This causes the slab indexes to be different on the gateway as on the EN. so we would not be able to directly compare the `(key,value)` but we can still compare the values but just getting them from the EN and the gateway and sorting both.

> 2. register related tools made for EN in the future can't be directly used by the gateway

True

> 3. mainnet database/snapshot can't be used by testnet by mistake, because the root owner is different.

The db contains other data besides the key value store that can discriminate between mainnet and testnet

> 4. we might want to compare the full set of registers with execution node to see if they are consistent. If the schema is the same, we could simply dump all registers to a file using the same tool, and compare them.

Same comment as on 1.

> 5. we could reuse the pruning module from flow-go without changing anything if we need one day.

True, but its easy enough to add as needed.

Another issue with reusing the `Registers` from flow-go is that the store function creates it s own batch and stores it immediately. In the gateway it would be great to not only store the registers, but also the block and transaction info in the same batch, so that they are always kept in sync and we don't need to think about the case when they are not.

I'm not sure it make sense to reuse the `Registers` from flow-go.

Maybe we can just get on a call about this.